gradle-baseline-java/src/test/groovy/com/palantir/baseline/BaselineClassUniquenessPluginIntegrationTest.groovy

```diff
@@ -0,0 +1,159 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline
+
+import java.nio.file.Files
+import java.util.stream.Stream
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.TaskOutcome
+
+class BaselineClassUniquenessPluginIntegrationTest extends AbstractPluginTest {
+
+    def standardBuildFile = """
+        plugins {
+            id 'java'
+            id 'com.palantir.baseline-class-uniqueness'
+        }
+        subprojects {
+            apply plugin: 'java'
+        }
+        repositories {
+            mavenCentral()
+            maven { url 'https://dl.bintray.com/palantir/releases' }
+        }
+    """.stripIndent()
+
+    def 'Task should run as part of :check'() {
+        when:
+        buildFile << standardBuildFile
+
+        then:
+        def result = with('check', '--stacktrace').build()
+        result.task(':checkClassUniqueness').outcome == TaskOutcome.SUCCESS
+    }
+
+    def 'detect duplicates in two external jars'() {
+        when:
+        buildFile << standardBuildFile
+        buildFile << """
+        dependencies {
+            compile group: 'javax.el', name: 'javax.el-api', version: '3.0.0'
+            compile group: 'javax.servlet.jsp', name: 'jsp-api', version: '2.1'
+        }   
+        """.stripIndent()
+        BuildResult result = with('checkClassUniqueness').buildAndFail()
+
+        then:
+        result.output.contains("26 Identically named classes with differing impls found in [javax.servlet.jsp:jsp-api:2.1, javax.el:javax.el-api:3.0.0]: [javax.")
+        result.getOutput().contains("'runtime' contains multiple copies of identically named classes")
+        result.getOutput().contains("(26 classes)  javax.servlet.jsp:jsp-api:2.1 javax.el:javax.el-api:3.0.0");
+        println result.getOutput()
+    }
+
+
+    def 'ignores duplicates when the implementations are identical'() {
+        when:
+        buildFile << standardBuildFile
+        buildFile << """
+        dependencies {
+            compile 'com.palantir.tritium:tritium-api:0.9.0'
+            compile 'com.palantir.tritium:tritium-core:0.9.0'
```

Comment:
what are the duplicate classes here? i'd expect tritium-core to pull in tritium-api and therefore not have any duplicate classes

Reply:
Just a package-info.java